src/coprocessor/codec/mysql/json/json_remove.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Json;
+use super::path_expr::{PathLeg, PathExpression};
+use super::super::Result;
+
+impl Json {
+    // Remove elements from Json,
+    // All path expressions cannot contain * or ** wildcard.
+    // If any error occurs, the input won't be changed.
+    pub fn remove(&mut self, path_expr_list: &[PathExpression]) -> Result<()> {
+        if path_expr_list.iter().any(|expr| expr.legs.is_empty() || expr.contains_any_asterisk()) {
+            return Err(box_err!("Invalid path expression"));
+        }
+
+        for expr in path_expr_list {
+            self.remove_path(&expr.legs);
```

Comment:
The comment above says if any error occurs, the input won't be changed. What if for path_expr_list [expr1, expr2, expr3], both expr1 and expr3 are valid, but expr2 is invalid?